sha/ta/include/sha_ta.h

```diff
@@ -0,0 +1,99 @@
+/* SPDX-License-Identifier: BSD-2-Clause */
+/*
+ * Copyright (c) 2025, Advanced Micro Devices, Inc. All rights reserved.
+ */
+
+#ifndef __SHA_TA_H__
+#define __SHA_TA_H__
+
+/* UUID of the SHA example trusted application */
+#define TA_SHA_UUID \
+	{ 0x1dc6a16b, 0x2fba, 0x4aa1, \
+		{ 0x95, 0x19, 0xea, 0x8a, 0x6c, 0x8c, 0x16, 0xe5 } }
+
+/*
+ * TA_SHA_CMD_PREPARE - Allocate resources for the SHA operation
+ * param[0] (value) a: TA_ALGO_HMAC_xxx, b: unused
+ * param[1] (value) a: key size in bytes, b: unused
+ * param[2] (value) a: obj_type, b: unused
+ * param[3] unused
+ */
+#define TA_SHA_CMD_PREPARE		0
+
+/* SHA Algo */
+#define TEE_ALG_SHA1			0x50000002
+#define TEE_ALG_SHA224                  0x50000003
+#define TEE_ALG_SHA256                  0x50000004
+#define TEE_ALG_SHA384                  0x50000005
+#define TEE_ALG_SHA512                  0x50000006
+#define TEE_ALG_SHA3_224                0x50000008
+#define TEE_ALG_SHA3_256                0x50000009
+#define TEE_ALG_SHA3_384                0x5000000A
+#define TEE_ALG_SHA3_512                0x5000000B
+#define TEE_ALG_HMAC_SHA1               0x30000002
+#define TEE_ALG_HMAC_SHA224             0x30000003
+#define TEE_ALG_HMAC_SHA256             0x30000004
+#define TEE_ALG_HMAC_SHA384             0x30000005
+#define TEE_ALG_HMAC_SHA512             0x30000006
+#define TEE_ALG_SHAKE128                0x50000101
+#define TEE_ALG_SHAKE256                0x50000102
+#define TEE_ALG_AES_CMAC                0x30000610
+
+/* Object types */
+#define TA_TYPE_HMAC_SHA256		0
+#define TA_TYPE_HMAC_SHA1		1
+#define TA_TYPE_HMAC_SHA224		2
+#define TA_TYPE_HMAC_SHA384		3
+#define TA_TYPE_HMAC_SHA512		4
+#define TA_TYPE_AES			5
+
+/* Algo Type */
+#define SHA_HMAC			0
+#define BASE_SHA			1
```

Comment:
See my comment about these IDs not being used by the TA hence not needed in the TA API header file.